dbms/src/Storages/MergeTree/MergeTreeDataWriter.cpp

```diff
@@ -75,12 +75,17 @@ void buildScatterSelector(
 }
 
 /// Computes ttls and updates ttl infos
-void updateTTL(const MergeTreeData::TTLEntry & ttl_entry, MergeTreeDataPart::TTLInfos & ttl_infos, Block & block, const String & column_name)
+void updateTTL(const MergeTreeData::TTLEntry & ttl_entry,
+    MergeTreeDataPart::TTLInfos & ttl_infos,
+    DB::MergeTreeDataPartTTLInfo & ttl_info,
+    Block & block, bool update_part_min_max_ttls)
 {
+    bool remove_column = false;
     if (!block.has(ttl_entry.result_column))
+    {
         ttl_entry.expression->execute(block);
-
-    auto & ttl_info = (column_name.empty() ? ttl_infos.table_ttl : ttl_infos.columns_ttl[column_name]);
+        remove_column = true;
```

Comment:
Why we remove columns after execution? Looks like side effect of this function. At least we need to update comment.

Reply:
Because we don't want to spoil block with extra columns. In this case merge would complain about extra columns (different table structure).

Reply:
So we just store expression result in block, than use to update ttl_info, and after that we remove it. OK.